Scripts/main/old/find_diff/diff_config_test.py.txt
5d4
< # config["chosen_machine"] = "google"
7,17c6
< config["base_folder_names"] = ["16012024_adam_datalossonly_infinite_continued", "16012024_adam_datalosscontloss_paraview_infinite", "19012024_adam_cont_boundary_RANS_loss_infinite"]
< # config["base_folder_names"] = ["16012024_adam_datalossonly_infinite_continued"]
< config["training"]["output_params"] = ['Pressure', 'Velocity:0', 'Velocity:1', 'Velocity:2', 'TurbVisc']
< config["training"]["output_params_modf"] = ['Pressure', 'Velocity_X', 'Velocity_Y', 'Velocity_Z', 'TurbVisc']
< config["train_test"]["test"] = True
< config["train_test"]["evaluate"] = True
< config["train_test"]["evaluate_new_angles"] = True
< config["plotting"]["make_logging_plots"] = True
< config["plotting"]["make_div_plots"] = True
< config["plotting"]["make_RANS_plots"] = True
< # config["plotting"]["make_data_plots"] = True
---
> ################################################################################################################################################################################################################################################################################################################################
19,21c8,18
< # config["base_folder_names"] = ["06012024_adam_datalossonly_infinite"]
< # config["training"]["output_params"] = ['Pressure', 'Velocity:0', 'Velocity:1', 'Velocity:2', 'TurbVisc']
< # config["training"]["output_params_modf"] = ['Pressure', 'Velocity_X', 'Velocity_Y', 'Velocity_Z', 'TurbVisc']
---
> # config["machine"] = {
> #         "mac": os.path.join(os.path.expanduser("~"), "Dropbox", "School", "Graduate", "CERN", "Temp_Files", "nonlineardynamics", "ameir_PINNs", "ladefense"),
> #         "CREATE": os.path.join('Z:\\', "ladefense"),
> #         "google": f"/content/drive/Othercomputers/MacMini/ladefense",
> #     }
> # config["data"]["geometry"] = "ladefense.stl"
> 
> # config["base_folder_names"] = ["16012024_adam_datalossonly_infinite", "07022024_adam_datalossonly_infinite_paraview_reduced_ladefense", "06022024_adam_datalossonly_infinite_A100"]
> # config["training"]["boundary"] = [[-2520,2520,-2520,2520,0,1000],100]
> # config["plotting"]["lim_min_max"] = [(-0.3, 0.3),(-0.3, 0.3),(0, 1)] 
> # config["plotting"]["plotting_params"] = [['X-Y',5,5]]
24c21
< # config["train_test"]["evaluate_new_angles"] = True
---
> # # config["train_test"]["evaluate_new_angles"] = True
26c23,24
< # config["plotting"]["make_div_plots"] = True
---
> # # config["plotting"]["make_div_plots"] = True
> # # config["plotting"]["make_RANS_plots"] = True
29c27,31
< # config["base_folder_names"] = ["04012024_adam_datalosscontloss_infinite_adaptiveweighting", "04012024_adam_datalossinletloss_infinite_adaptiveweighting","04012024_adam_datalossnosliploss_infinite_adaptiveweighting"]
---
> # config["base_folder_names"] = ["07022024_adam_datalossonly_infinite_paraview_reduced_ladefense"]
> # config["data"]["startname_data"] = "Paraview_Reduced_CFD"
> # config["training"]["boundary"] = [[-2520,2520,-2520,2520,0,1000],100]
> # config["plotting"]["lim_min_max"] = [(-0.3, 0.3),(-0.3, 0.3),(0, 1)]
> # config["plotting"]["plotting_params"] = [['X-Y',5,5]]
34,35c36,38
< # config["plotting"]["make_div_plots"] = True
< # config["plotting"]["make_data_plots"] = True
---
> # # config["plotting"]["make_div_plots"] = True
> # # config["plotting"]["make_RANS_plots"] = True
> # # config["plotting"]["make_data_plots"] = True
37c40,44
< # config["base_folder_names"] = ["01012024_adam_datalossonly_infinite_tpu"]
---
> # config["base_folder_names"] = ["30012024_adam_datalossonly_infinite_reduced"]
> # config["data"]["startname_data"] = "Reduced_CFD"
> # config["training"]["boundary"] = [[-504,504,-504,504,0,300],100]
> # config["plotting"]["lim_min_max"] = [(-1, 1),(-1, 1),(0, 1)] 
> # config["plotting"]["plotting_params"] = [['X-Y',5,5]]
40c47
< # config["train_test"]["evaluate_new_angles"] = True
---
> # # config["train_test"]["evaluate_new_angles"] = True
41a49,62
> # # config["plotting"]["make_div_plots"] = True
> # # config["plotting"]["make_RANS_plots"] = True
> # # config["plotting"]["make_data_plots"] = True
> 
> ################################################################################################################################################################################################################################################################################################################################
> 
> # config["base_folder_names"] = ["09022024_adam_paraviewreduceddataloss_allphysicsloss_infinite"]
> # config["data"]["startname_data"] = "Paraview_Reduced_CFD"
> # config["training"]["output_params"] = ['Pressure', 'Velocity:0', 'Velocity:1', 'Velocity:2', 'TurbVisc']
> # config["training"]["output_params_modf"] = ['Pressure', 'Velocity_X', 'Velocity_Y', 'Velocity_Z', 'TurbVisc']
> # config["train_test"]["test"] = True
> # config["train_test"]["evaluate"] = True
> # # config["train_test"]["evaluate_new_angles"] = True
> # config["plotting"]["make_logging_plots"] = True
43c64,65
< # config["plotting"]["make_data_plots"] = True
---
> # config["plotting"]["make_RANS_plots"] = True
> # # config["plotting"]["make_data_plots"] = True
45c67,70
< # config["base_folder_names"] = ["18122023_adam_datalosscontloss_infinite_adaptiveweighting", "18122023_adam_datalossinletloss_infinite_adaptiveweighting","18122023_adam_datalossnosliploss_infinite_adaptiveweighting"]
---
> # config["base_folder_names"] = ["06012024_adam_datalossonly_infinite", "16012024_adam_datalossonly_infinite_continued_test"]
> config["base_folder_names"] = ["16012024_adam_datalossonly_infinite_continued_test"]
> config["training"]["output_params"] = ['Pressure', 'Velocity:0', 'Velocity:1', 'Velocity:2', 'TurbVisc']
> config["training"]["output_params_modf"] = ['Pressure', 'Velocity_X', 'Velocity_Y', 'Velocity_Z', 'TurbVisc']
49c74,77
< # config["plotting"]["make_logging_plots"] = True
---
> config["train_test"]["boundary_test"] = True
> config["plotting"]["lim_min_max"] = [(-0.05, 0.05),(-0.1, 0.3),(0, 0.3)]
> config["plotting"]["plotting_params"] = [['X-Y',50,5],['Y-Z',500,5]]
> config["plotting"]["make_logging_plots"] = True
50a79
> # config["plotting"]["make_RANS_plots"] = True
53,67c82
< # config["base_folder_names"] = ["test_4"]
< # config["train_test"]["train"] = True
< # config["training"]["use_epochs"] = True
< # config["training"]["num_epochs"] = 100
< # config["training"]["print_epochs"] = 1
< # config["loss_components"]["data_loss"] = True
< # config["loss_components"]["cont_loss"] = True
< # config["loss_components"]["boundary_loss"] = True
< # config["loss_components"]["no_slip_loss"] = True
< # config["loss_components"]["inlet_loss"] = True
< # config["loss_components"]["use_weighting"] = True
< # config["train_test"]["test_size"] = 0.99999
< # config["training"]["use_batches"] = True
< # config["training"]["batch_size"] = 1
< # config["train_test"]["data_loss_test_size"] = 0.9
---
> ################################################################################################################################################################################################################################################################################################################################
69,75d83
< # config["base_folder_names"] = ["26112023_adam_datalossonly_infinite"]
< # config["training"]["activation_function"] = nn.ReLU
< # config["train_test"]["make_div_plots"] = True
< # # config["train_test"]["new_angles"] = [0]
< # # config["train_test"]["new_angles"] = config["training"]["all_angles"]
< # # config["train_test"]["new_angles_test_size"] = 0.1
< 


Scripts/main/current/done/config.py
from definitions import * 

config = {
    "lbfgs_optimizer": {
        "type": "LBFGS",
        "learning_rate": 0.00001,
        "max_iter": 200000,
        "max_eval": 50000,
        "history_size": 50,
        "tolerance_grad": 1e-05,
        "tolerance_change": 0.5 * np.finfo(float).eps,
        "line_search_fn": "strong_wolfe"
    },
    "adam_optimizer": {
        "type": "Adam",
        "learning_rate": 0.001,
    },
    "training": {
        "number_of_hidden_layers": 4,
        "neuron_number": 128,
        "input_params": ['Points:0', 'Points:1', 'Points:2', 'cos(WindAngle)', 'sin(WindAngle)'], 
        "input_params_modf": ["X", "Y", "Z", "cos(WindAngle)", "sin(WindAngle)"], 
        "output_params": ['Velocity:0', 'Velocity:1', 'Velocity:2'], #['Pressure', 'Velocity:0', 'Velocity:1', 'Velocity:2', 'TurbVisc']
        "output_params_modf": ['Velocity_X', 'Velocity_Y', 'Velocity_Z'], #['Pressure', 'Velocity_X', 'Velocity_Y', 'Velocity_Z', 'TurbVisc']
        "activation_function": nn.ELU, #nn.ReLU, nn.Tanh,
        "batch_normalization": False,
        "dropout_rate": None,
        "use_epochs": False,
        "num_epochs": 1,
        "print_epochs": 10,
        "use_batches": False,
        "batch_size": 2**15,
        "use_data_for_cont_loss": False,
        "angles_to_leave_out": [135],
        "angles_to_train": [0,15,30,45,60,75,90,105,120,150,165,180],
        "all_angles": [0,15,30,45,60,75,90,105,120,135,150,165,180],
        "boundary": [[0,1000,0,1000,0,1000],100], #[[-2520,2520,-2520,2520,0,1000],100]
        "loss_diff_threshold": 1e-5,
        "consecutive_count_threshold": 10,
        "feature_scaler": sklearn.preprocessing.StandardScaler(), #sklearn.preprocessing.MinMaxScaler(feature_range=(-1,1))
        "target_scaler": sklearn.preprocessing.StandardScaler(), #sklearn.preprocessing.MinMaxScaler(feature_range=(-1,1))
    },
    "machine": {
        "mac": os.path.join(os.path.expanduser("~"), "Dropbox", "School", "Graduate", "CERN", "Temp_Files", "nonlineardynamics", "ameir_PINNs", "cylinder_cell"),
        "CREATE": os.path.join('Z:\\', "cylinder_cell"),
        "google": f"/content/drive/Othercomputers/MacMini/cylinder_cell",
    },
    "data": {
        "density": 1,
        "kinematic_viscosity": 1e-5,
        "data_folder_name": 'data',
        "extension": '.csv',
        "startname_data": 'CFD',
        "startname_meteo": 'meteo_',
        "output_zip_file": 'output.zip',
        "geometry": 'scaled_cylinder_sphere.stl' #'ladefense.stl'
    },
    "loss_components": {
        "data_loss": False,
        "cont_loss": False,
        "momentum_loss": False,
        "no_slip_loss":False, 
        "inlet_loss": False,
        "use_weighting": False,
        "weighting_scheme": 'adaptive_weighting', #'gradient_magnitude'
        "adaptive_weighting_initial_weight": 0.9,
        "adaptive_weighting_final_weight": 0.1,
    },
    "train_test": {
        "train": False,
        "test": False,
        "evaluate": False,
        "evaluate_new_angles": False,
        "test_size": 0.1,
        "data_loss_test_size": None, #10000 
        "new_angles_test_size": 0.99999,
        "random_state": 42,
        "new_angles": [0,15,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,75,90,105,120,135,150,165,180]
    },
    "plotting": {
        "lim_min_max": [(-1, 1),(-1, 1),(0, 1)], #[(-0.3, 0.3),(-0.3, 0.3),(0, 1)]
        "arrow": [False, [[500,500],[500,570]]],
        "plotting_params": [['X-Y',50,5]], # [['X-Z',570,5],['Y-Z',500,5],['X-Y',50,5]] # ['X-Y',5,5] # [['X-Z',-300,5],['Y-Z',-300,5],['X-Y',5,5]]
        "make_logging_plots": False,
        "make_data_plots": False,
        "make_div_plots": False,
        "make_RANS_plots": False,
    },
    "chosen_machine": "mac",
    "chosen_optimizer": "adam_optimizer",
    "base_folder_names": "test",
    "base_folder_names": ["test"]
}